opal/mca/btl/tcp/btl_tcp_component.c

```diff
@@ -1300,44 +1336,133 @@ static void mca_btl_tcp_component_recv_handler(int sd, short flags, void* user)
     opal_socklen_t addr_len = sizeof(addr);
     mca_btl_tcp_proc_t* btl_proc;
     int retval;
+    char str[128];
+    size_t len = strlen(mca_btl_tcp_magic_id_string);
 
     OBJ_RELEASE(event);
 
+    /* Receive the magic string */
+    assert(len < sizeof(str));
+    str[0] = '\0';
+    /* TODO: recv_blocking() will block forever, without timeout
```

Comment:
Can setting the SO_RCVTIMEO temporary during mca_btl_tcp_recv_blocking be the solution ?